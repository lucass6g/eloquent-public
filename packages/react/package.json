{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
