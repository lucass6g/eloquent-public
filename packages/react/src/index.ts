export {Switch, Case, Default} from './components/conditional-switch'