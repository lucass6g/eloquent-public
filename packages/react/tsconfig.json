{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
