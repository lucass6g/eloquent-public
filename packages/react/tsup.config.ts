import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})