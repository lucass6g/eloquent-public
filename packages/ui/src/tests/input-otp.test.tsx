import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { InputOTP } from "..";
import userEvent from "@testing-library/user-event";

describe("Input OTP", () => {
  it("should match the snapshot", () => {
    const component = render(
      <InputOTP.Root maxLength={4}>
        <InputOTP.Group>
          <InputOTP.Slot index={0} />
          <InputOTP.Slot index={1} />
        </InputOTP.Group>
        <InputOTP.Separator />
        <InputOTP.Group>
          <InputOTP.Slot index={2} />
          <InputOTP.Slot index={3} />
        </InputOTP.Group>
      </InputOTP.Root>
    );

    expect(component.baseElement).toMatchSnapshot();
  });

  it("should be filled out correctly", () => {
    const stateFn = vi.fn();

    render(
      <InputOTP.Root maxLength={4} onChange={stateFn}>
        <InputOTP.Group>
          <InputOTP.Slot index={0} />
          <InputOTP.Slot index={1} />
        </InputOTP.Group>
        <InputOTP.Separator />
        <InputOTP.Group>
          <InputOTP.Slot index={2} />
          <InputOTP.Slot index={3} />
        </InputOTP.Group>
      </InputOTP.Root>
    );

    const input = screen.getByRole("textbox");
    fireEvent.change(input, { target: { value: "1234" } });

    const filledInput = screen.queryByDisplayValue("1234");
    expect(filledInput).toBeInTheDocument();

    expect(stateFn).toHaveBeenCalled();
  });

  it("should be disabled", async () => {
    render(
      <InputOTP.Root disabled maxLength={4}>
        <InputOTP.Group>
          <InputOTP.Slot index={0} />
          <InputOTP.Slot index={1} />
        </InputOTP.Group>
        <InputOTP.Separator />
        <InputOTP.Group>
          <InputOTP.Slot index={2} />
          <InputOTP.Slot index={3} />
        </InputOTP.Group>
      </InputOTP.Root>
    );

    const input = screen.getByRole("textbox");
    await waitFor(() => userEvent.type(input, "1234"));

    const filledInput = screen.queryByDisplayValue("1234");
    expect(filledInput).not.toBeInTheDocument();

    expect(input).toBeDisabled();
  });
});
