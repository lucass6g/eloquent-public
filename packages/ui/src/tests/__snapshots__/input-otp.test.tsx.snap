// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input OTP > should match the snapshot 1`] = `
<body>
  <div>
    <noscript />
    <div
      class="flex items-center gap-2 has-[:disabled]:opacity-50"
      data-input-otp-container="true"
      style="position: relative; cursor: text; user-select: none; pointer-events: none; --root-height: 0px;"
    >
      <div
        class="flex items-center"
      >
        <div
          class="relative flex h-9 w-9 items-center justify-center border-y border-r border-input text-sm shadow-sm transition-all first:rounded-l-md first:border-l last:rounded-r-md"
        />
        <div
          class="relative flex h-9 w-9 items-center justify-center border-y border-r border-input text-sm shadow-sm transition-all first:rounded-l-md first:border-l last:rounded-r-md"
        />
      </div>
      <div
        role="separator"
      >
        <svg
          fill="none"
          height="15"
          viewBox="0 0 15 15"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M5 7.5C5 7.22386 5.22386 7 5.5 7H9.5C9.77614 7 10 7.22386 10 7.5C10 7.77614 9.77614 8 9.5 8H5.5C5.22386 8 5 7.77614 5 7.5Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="flex items-center"
      >
        <div
          class="relative flex h-9 w-9 items-center justify-center border-y border-r border-input text-sm shadow-sm transition-all first:rounded-l-md first:border-l last:rounded-r-md"
        />
        <div
          class="relative flex h-9 w-9 items-center justify-center border-y border-r border-input text-sm shadow-sm transition-all first:rounded-l-md first:border-l last:rounded-r-md"
        />
      </div>
      <div
        style="position: absolute; inset: 0; pointer-events: none;"
      >
        <input
          autocomplete="one-time-code"
          class="disabled:cursor-not-allowed"
          data-input-otp="true"
          inputmode="numeric"
          maxlength="4"
          pattern="^\\d+$"
          style="position: absolute; inset: 0; width: 100%; height: 100%; display: flex; text-align: left; opacity: 1; color: transparent; pointer-events: all; background: transparent; caret-color: transparent; border: 0px solid transparent; outline: 0 solid transparent; box-shadow: none; line-height: 1; letter-spacing: -.5em; font-family: monospace; font-variant-numeric: tabular-nums;"
          value=""
        />
      </div>
    </div>
  </div>
</body>
`;
